libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -152,6 +152,39 @@ string YulUtilFunctions::storeLiteralInMemoryFunction(string const& _literal)
 	});
 }
 
+string YulUtilFunctions::copyLiteralToStorageFunction(string const& _literal)
+{
+	string functionName = "copy_literal_to_storage_" + util::toHex(util::keccak256(_literal).asBytes());
+
+	return m_functionCollector.createFunction(functionName, [&]() {
+		size_t words = (_literal.length() + 31) / 32;
+		vector<map<string, string>> wordParams(words);
+		for (size_t i = 0; i < words; ++i)
+		{
+			wordParams[i]["offset"] = to_string(i);
+			wordParams[i]["wordValue"] = formatAsStringOrNumber(_literal.substr(32 * i, 32));
+		}
+
+		return Whiskers(R"(
+			function <functionName>(slot) {
```

Comment:
For cases like this I'm wondering *every* time, if it wasn't easier to read to split them up. Here for example split into a `long` and a `short` case with two templates... I'm not saying we should necessarily do that here, but I would be curious about the general opinion: is using one template with more case variables really preferable to two templates (maybe as `<body>` in the outer function template)?
I personally think I would have review a split version twice as fast...

Reply:
Hm, indeed, you are right. It is not easiest to review as is! Also, it looks much simpler if I split into different cases